### PR TITLE
feat: add deploy command to simplify deployment

### PR DESCRIPTION
**Pull request by dpportet**
# Description

* Add a deploy command which ensures dependencies are installed, workspace is clean 
* Removed the `rmraf node_modules` since clean should just delete the build folders

## Type of change
- [X] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?
- [X] Did a clean deployment

# Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes

## Legal

This project is available under the [Apache 2.0 License](http://www.apache.org/licenses/LICENSE-2.0.html).

### CODE REVIEW

File: deploymentguide/README.md [hunk 1]
```diff
@@ -23,14 +23,11 @@ This will deploy the application to your AWS account using CDK.
...
-   ```sh
-   yarn install
-   ```
-1. Deploy the Core service and resource dependencies:
+1. Deploy the application using the deploy command:
    ```sh
-   yarn workspace cdk cdk deploy --all
+   yarn deploy
    ```
+1. This will install dependencies and deploy the application to the cloud using CDK and creating CloudFormation stacks that support the aplpication.
```
Comment (tracy-french) [inline deploymentguide/README.md:30]: Typo! aplpication

**Review by corteggiano — APPROVED** [review]

**Review by chejimmy — COMMENTED** [review]
Please update the [`clean`](https://github.com/awslabs/iot-application/blob/main/packages/eslint-config-custom/package.json#L7) under [eslint-config-custom](https://github.com/awslabs/iot-application/tree/main/packages/eslint-config-custom) as well to be consistent 🙏

**Comment by chejimmy** [comment]
> Please update the [`clean`](https://github.com/awslabs/iot-application/blob/main/packages/eslint-config-custom/package.json#L7) under [eslint-config-custom](https://github.com/awslabs/iot-application/tree/main/packages/eslint-config-custom) as well to be consistent 🙏

dismiss this, I think I just missed it, it is [here](https://github.com/awslabs/iot-application/blob/436ce638739cda0a4a10b210a37f587630ec2a7e/packages/eslint-config-custom/package.json#L7)

**Review by chejimmy — DISMISSED** [review]
LGTM 🙏

**Review by tracy-french — CHANGES_REQUESTED** [review]

**Review by tracy-french — APPROVED** [review]

**Review by chejimmy — APPROVED** [review]

**Pull request merged**